GeosCore/calc_met_mod.F90

```diff
@@ -293,24 +293,23 @@ SUBROUTINE AIRQNT( Input_Opt, State_Chm, State_Grid, State_Met, &
        IsLocNoon(I,J) = ( LocTimeSec(I,J)          <= 43200  .and. &
                           LocTimeSec(I,J) + Dt_Sec >= 43200 )
 
+       ! Snow
+       FRSNOW = State_Met%FRSNOW(I,J)
+
        ! Land without snow or ice
-       FRLAND_NOSNO_NOICE = State_Met%FRLAND(I,J) -                &
-               (State_Met%FRLAND(I,J)*State_Met%FRSNO(I,J))
+       FRLAND_NOSNOW_NOICE = State_Met%FRLAND(I,J) - State_Met%FRSNOW(I,J)
 
        ! Water without sea ice
        FRWATER = State_Met%FRLAKE(I,J) + State_Met%FROCEAN(I,J) - State_Met%FRSEAICE(I,J)
       
        ! Land ice and sea ice
-       FRICE = State_Met%FRLANDIC(I,J) + State_Met%FRSEAICE(I,J)
-      
-       ! Snow
-       FRSNO = State_Met%FRSNO(I,J)
-      
+       FRICE = State_Met%FRLANDICE(I,J) + State_Met%FRSEAICE(I,J)
+
        ! Set IsLand, IsWater, IsIce, IsSnow based on max fractional area
-       State_Met%IsLand(I,J)  = (FRLAND_NOSNO_NOICE > MAX(FRWATER, FRICE, FRSNO))
-       State_Met%IsWater(I,J) = (FRWATER > MAX(FRLAND_NOSNO_NOICE, FRICE, FRSNO))
-       State_Met%IsIce(I,J)   = (FRICE > MAX(FRLAND_NOSNO_NOICE, FRWATER, FRSNO))
-       State_Met%IsSnow(I,J)  = (FRSNO > MAX(FRLAND_NOSNO_NOICE, FRWATER, FRICE))
+       State_Met%IsLand(I,J)  = (FRLAND_NOSNOW_NOICE > MAX(FRWATER, FRICE, FRSNOW))
```

Comment:
Shouldn't the IsLand logical include snow-covered land? Therefore should this line instead be:

```
State_Met%IsLand(I,J) = (FRLAND > MAX(FRWATER, FRICE))
```

Reply:
The intention is for land, ice, water, and snow to be separate categories. This met-field only appears in the code in land_mercury_mod.F90, in a commented out subroutine. So perhaps we can delete it to avoid confusion.

Reply:
I think that might be best -- otherwise we'd need to rename it to something like IsLand_NoSnow to be clear

Reply:
Actually, I just found it used in drydep_mod. Since these four logical masks are supposed to mutually exclusive (only one true for each grid box), I could create an index variable similar to how we used to use LWI.